File: dotnet-adapters-shortest-paths-core/main/java/com/programmerare/shortestpaths/core/impl/PathImpl.java
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
package com.programmerare.shortestpaths.core.impl;

import java.util.List;

import com.programmerare.shortestpaths.core.api.Edge;
import com.programmerare.shortestpaths.core.api.Path;
import com.programmerare.shortestpaths.core.api.Vertex;
import com.programmerare.shortestpaths.core.api.Weight;
import com.programmerare.shortestpaths.core.impl.generics.PathGenericsImpl;

public class PathImpl extends PathGenericsImpl<Edge , Vertex , Weight> implements Path {

	protected PathImpl(final Weight totalWeight, final List<Edge> edges) {
		super(totalWeight, edges);
	}

	public static Path createPath(final Weight totalWeight, final List<Edge> edges) {
		return new PathImpl(totalWeight, edges);
	}

}

File: dotnet-adapters-shortest-paths-core/main/com.programmerare.shortestpaths/core/impl/PathImpl.cs
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
using com.programmerare.shortestpaths.core.api;
using com.programmerare.shortestpaths.core.impl.generics;
using System.Collections.Generic;

namespace com.programmerare.shortestpaths.core.impl
{
    public class PathImpl : PathGenericsImpl<Edge, Vertex, Weight> , Path 
    {

	    protected PathImpl(Weight totalWeight, IList<Edge> edges): base(totalWeight, edges) {
		
	    }

	    public static Path createPath(Weight totalWeight, IList<Edge> edges) {
		    return new PathImpl(totalWeight, edges);
	    }

    }
}